Wikipedia/Code/ArticleViewController.swift

```diff
@@ -67,7 +66,13 @@ class ArticleViewController: ViewController, HintPresenting {
         return rc
     }()
     
-    @objc init?(articleURL: URL, dataStore: MWKDataStore, theme: Theme) {
+    @objc convenience init?(articleURL: URL, dataStore: MWKDataStore, theme: Theme) {
```

Comment:
As far as I can tell, production uses of `SchemeHandler` are always via its shared class instance. I wonder if there's value in removing this convenience initializer and just being explicit for the Objective-C calls (suggestion here: https://github.com/wikimedia/wikipedia-ios/commit/5730f5bb08b59032a927d75b75fb0bdaea7e2b01). This way, regardless of if it's via dependency injection for unit testing or by production, there's always only one initialization path into `ArticleViewController`.

Only a suggestion – I can see it either way if you feel more strongly about this approach.

Reply:
That sounds like a good change - this is done.